src/scripts/main.js

```diff
@@ -1,3 +1,401 @@
+    const select = document.createElement('select');
+
+    select.className = 'cell-input';
+
+    select.insertAdjacentHTML('beforeend', `
+      <option value="Tokyo">Tokyo</option>
+      <option value="Singapore">Singapore</option>
+      <option value="London">London</option>
+      <option value="New York">New York</option>
+      <option value="Edinburgh">Edinburgh</option>
+      <option value="San Francisco">San Francisco</option>
+    `);
+
+    item.textContent = '';
+    item.appendChild(select);
+    select.focus();
+
+    select.addEventListener('change', ev => {
+      updateCellCurrent(item, select);
+    });
+
+    select.addEventListener('blur', ev => {
+      updateCellInitial(item, currentItem, select);
+    });
+
+    select.addEventListener('keydown', ev => {
+      if (ev.key === 'Enter') {
+        updateCellCurrent(item, select);
+      }
+    });
+  }
+
+  if (!item.querySelector('input') && targetCell > 2) {
+    const input = document.createElement('input');
+
+    item.className = 'cell-input';
+    item.textContent = '';
+    input.className = 'cell-input';
+    input.value = item.textContent;
+    input.type = 'number';
+
+    item.appendChild(input);
+    input.focus();
+
+    const TARGET_CELL_AGE = 3;
+    const TARGET_CELL_CURRENCY = 4;
+
+    switch (targetCell) {
+      case TARGET_CELL_AGE: {
+        input.className = 'age';
+
+        input.addEventListener('keydown', ev => {
+          if (ev.key === 'Enter') {
+            if ((input.value < 18 || input.value > 90)) {
+              pushNotification(...errorAge);
+
+              updateCellInitial(item, currentItem, input);
+
+              return;
+            };
+
+            updateCellCurrent(item, input);
+          }
+        });
+
+        input.addEventListener('blur', ev => {
+          if (ev.target.value === '') {
+            ev.target.value = currentItem;
+          }
+
+          if (input.value < 18 || input.value > 90) {
+            pushNotification(...errorAge);
+
+            updateCellInitial(item, currentItem, input);
+
+            return;
+          };
+
+          updateCellCurrent(item, input);
+        });
+
+        break;
+      }
+
+      case TARGET_CELL_CURRENCY: {
+        input.addEventListener('keydown', ev => {
+          if (ev.key === 'Enter') {
+            if (input.value.length === 0) {
+              updateCellInitial(item, currentItem, input);
+
+              return;
+            };
+
+            item.textContent = formatCurrency(input.value);
+            input.remove();
+          }
+        });
+
+        input.addEventListener('blur', ev => {
+          if (!ev.target.value) {
+            ev.target.value = currentItem;
+          }
+
+          if (input.value.length === 0) {
+            updateCellInitial(item, currentItem, input);
+
+            return;
+          };
+
+          item.textContent = formatCurrency(input.value);
+          input.remove();
```

Comment:
in all cases on `blur` event you're doing the same as on `Enter` `keydown` + one additional `if` block
would be nice to reuse this code as well to avoid duplicating

Reply:
In the IF block, I already use a function to avoid duplication.

Everything else is fixed

...
updateCellInitial(item, currentItem, input);
...

...
function updateCellInitial(item, currentItem, input) {
  item.textContent = currentItem;
  input.remove();
}